pre_award/assessment_store/db/models/assessment_record/assessment_records.py

```diff
@@ -71,6 +71,7 @@ class AssessmentRecord(BaseModel):
         "AssessmentFlag",
         primaryjoin="and_(AssessmentFlag.application_id == AssessmentRecord.application_id, "
         "AssessmentFlag.is_change_request == True)",
+        viewonly=True,  # this is a subset relationship; the unfiltered `flags` should be used for writes/deletes
```

Comment:
If my understanding is correct.. I would rather go with `overlaps` as it's allowing both `flags` relationship and `change_requests` relationship to update `Flag` models. They both point to the same Python model, I think it should be fine.

We could make it `read_only` to put the warning to rest. But I just find it an unexpected behaviour.

I'd even rather go for:
```python
@property
def change_requests(self):
    return self.flags.filter_by(is_change_request=True).all()
```

Reply:
I'd like to explore what happens when using that property because I suspect it will in most (all?) Cases require another trip to the DB. Would be nice to not need that.

Reply:
Yes. It would make another trip to the DB. Unless self.flag was not already used before and is loaded in the memory. 
That behaviour can be changed by loading flags together with Assessents, but then we might pull those in memory even when not needed.

I actually think `override` is the way. I am not sure I see the benefit for `viewonly`, and I can absolutely see future us trying to edit `assessment_record.change_requests` and spend 3 hours confused why it won't work. 

The `property` option is indeed less efficient, but at least it's easier to understand.